### DISCUSSION TITLE
Intercept H3 Response Body (Plus Types?)

### DISCUSSION THREAD

**Discussion by tobychidi** [nitrojs/nitro#1392] [Q&A] [answered]
Now, the plugin I want to create requires I append something to the body of the response.

Here is what was done in express. How do I replicate in H3/Nitro

const originalSend = res.send;
   res.send = function sendOverWrite(body) {
      originalSend.call(this, body);
      this._treblleResponsebody = body;
   };

**Comment by Hebilicious** [comment]
Would the `render:response` hook work for your use-case https://nuxt.com/docs/api/advanced/hooks#nitro-app-hooks-runtime-server-side work for you ?

**Reply by tobychidi** [reply]
`render:response` seems to only works for page renders and not for API requests.

**Reply by Hebilicious** [reply, marked as answer]
I think that your only option is a middleware, altough, they run globally before your logic. We do have some features coming that will enable the ability to run code after an event handler has returned, but in the meantime I also added a proposal that could allow you to do that https://github.com/unjs/nitro/issues/1396

However, if you need to do this now, you can do it with a custom nitro preset https://github.com/unjs/nitro/issues/1373

**Comment by rs-duongnt1** [comment]
You can try with nitro:
```typescript
export default defineNitroPlugin((nitroApp) => {
  nitroApp.hooks.hook("request", (event) => {
    console.log("on request", event.path);
  });

  nitroApp.hooks.hook("beforeResponse", (event, { body }) => {
    send(
      event,
      JSON.stringify({
        status: 200,
        data: body,
      })
    );
  });
});

```
